pkg/plugin/plugin.go

```diff
@@ -387,26 +387,88 @@ func ScanRootForDevices(nxGzipEnabled bool) ([]string, error) {
 	// relies on GHW_CHROOT=/host/dev
 	// lsblk -f --json --paths -s | jq -r '.blockdevices[] | select(.fstype != "xfs")' | grep mpath | grep -v fstype | sort -u | wc -l
 	// This may be the best way to get the devices.
+	config, err := loadDevicePluginConfig()
+	if err != nil {
+		klog.Warningf("ScanRootForDevices: failed to load config, proceeding with default behavior: %v", err)
+		config = &api.DevicePluginConfig{}
+	}
+
 	block, err := ghw.Block()
 	if err != nil {
 		fmt.Printf("Error getting block storage info: %v", err)
 		return nil, err
 	}
 	devices := []string{}
 	fmt.Printf("DEVICE: %v\n", block)
+
 	for _, disk := range block.Disks {
 		fmt.Printf("    - DISK: %v\n", disk.Name)
 		for _, part := range disk.Partitions {
 			fmt.Printf("        - PART: %v\n", part.Disk.Name)
-			devices = append(devices, part.Name)
+			devices = append(devices, "/dev/"+part.Name)
 		}
-		devices = append(devices, disk.Name)
+		devices = append(devices, "/dev/"+disk.Name)
 	}
+
 	if nxGzipEnabled {
-		devices = append(devices, nxGzipDevicePath)
+		devices = append(devices, "/dev/"+nxGzipDevicePath)
 		klog.Infof("nx-gzip enabled: appended /dev/crypto/nx-gzip to devices")
 	}
-	return devices, nil
+
+	filtered := []string{}
+
+	// Apply exclude patterns first
+	for _, dev := range devices {
+		if matchesAny(dev, config.ExcludeDevices) {
+			klog.Infof("Excluding device: %s", dev)
+			continue
+		}
+		filtered = append(filtered, dev)
+	}
+
+	finalDevices := []string{}
+
+	// Only include devices that match the include patterns and exist on the host.
+	useInclude := false
+	for _, item := range config.IncludeDevices {
+		if strings.TrimSpace(item) != "" {
+			useInclude = true
+			break
+		}
+
+		if item == "" {
+			klog.Warningf("Include-devices contains an empty string. Ignoring include-devices and proceeding with default behavior.")
+		}
+	}
+
+	if useInclude {
+		klog.Infof("Include-devices specified, overriding with: %v", config.IncludeDevices)
+		for _, pattern := range config.IncludeDevices {
+			if strings.TrimSpace(pattern) == "" {
+				continue
+			}
+			matches, err := filepath.Glob(pattern)
+			if err != nil {
+				klog.Warningf("Invalid include pattern: %s, skipping. Error: %v", pattern, err)
+				continue
+			}
+			for _, dev := range matches {
+				if _, err := os.Stat(dev); err == nil {
+					finalDevices = append(finalDevices, strings.TrimPrefix(dev, "/dev/"))
+					klog.Infof("Included device: %s", dev)
```

Comment:
```suggestion
					klog.V(4).Infof("Included device: %s", dev)
```
We should probably tune it to output less data.
We can get the full include data at the end with L470